indexers/keyvalue/RedisDBIndexer/__init__.py

```diff
@@ -0,0 +1,73 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import json
+from typing import Optional
+
+from jina.executors.indexers.keyvalue import BinaryPbIndexer
+from google.protobuf.json_format import Parse
+from jina.proto import jina_pb2
+
+
+class RedisDBIndexer(BinaryPbIndexer):
+    """
+    :class:`RedisDBIndexer` Use Redis as a key-value indexer.
+    """
+
+    def __init__(self,
+                 hostname: str = '0.0.0.0',
+                 port: int = 63079,
+                 db: int = 0,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.hostname = hostname
+        self.port = port
+        self.db = db
+
+    def get_add_handler(self):
+        """Get the database handler
+
+        """
+        import redis
+        r = redis.Redis(host=self.hostname, port=self.port, db=self.db, socket_timeout=None)
+        try:
+            r.ping()
+            return r
+        except redis.exceptions.ConnectionError as r_con_error:
+            self.logger.error('Redis connection error: ', r_con_error)
+
+
+    def add(self, objs):
```

Comment:
We've changed the driver `exec_fn` invocation for `KVIndexDriver`, now it takes `keys` & `values`, instead of `objs -> Dict`. This makes the indexer function properly on its own, but not with a `Flow`